lib/rack/test.rb

```diff
@@ -240,6 +240,7 @@ def env_for(path, env)
       def process_request(uri, env)
         uri = URI.parse(uri)
         uri.host ||= @default_host
+        uri.scheme ||= "https" if env["HTTPS"] == "on"
```

Comment:
My memory is a little vague around the details of this change. My guess is that this follows an already defined convention when testing with SSL. Ideally you could support both versions e.g. `%w(on true).include?(env["HTTPS"])`. However I would delegate that decision to the maintainers, there is probably good reason for using "on" that I'm not aware of.